### DISCUSSION TITLE
Editor lags when scrubbing or scrolling fast (related to Bass seek)

### DISCUSSION THREAD

**Discussion by Susko3** [ppy/osu#16356] [Q&A] [answered]
To repro:

1. open the editor with a song
2. (optional) unlock the scroll wheel if you have a Logitech mouse
3. scroll
4. alternatively, scrub through the timeline
 
Only really happens in multi-threaded mode. In single-threaded, the lag is minimal if the track is paused, and there is slightly lag if the track is playing. The lag is feels worse if the window is unfocused.

If this line ([TrackBass.cs#L278](https://github.com/ppy/osu-framework/blob/050a0b8639c9bd723100288a53923547ce87d487/osu.Framework/Audio/Track/TrackBass.cs#L278
)) is commented-out, then there is no lag.

If you play the song, unfocus the window, and then scroll fast, you can hear the track seeking while the game is 0 FPS.

Seems like track seeking is abused by `EditorClock`'s `TransformSeek`.

Scrolling:

https://user-images.githubusercontent.com/16479013/148622130-c6d6fee8-52e2-4084-a230-b4447535c66a.mp4

Scrubbing through the timeline:

https://user-images.githubusercontent.com/16479013/148624106-5019d9a3-726a-4f78-b08e-b507c2b0b4a7.mp4

[database.log](https://github.com/ppy/osu/files/7831991/database.log)
[network.log](https://github.com/ppy/osu/files/7831992/network.log)
[performance.log](https://github.com/ppy/osu/files/7831993/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7831994/runtime.log)

**Comment by peppy** [comment]
This *must* have regressed. It was already fixed once before. Potentially related to audio mixer changes.

**Reply by Susko3** [reply]
https://github.com/ppy/osu/issues/14515

But doesn't look much related. Only that `ChannelSetPosition` is called too often.

**Reply by bdach** [reply]
can we not double up discussion now that there is an issue for this?

**Comment by bdach** [comment, marked as answer]
moved to #16360
